From 789582b940a7591a7234b682f6b4342059041a47 Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Sun, 25 Oct 2026 21:39:31 +0100
Subject: [PATCH 1/1] Shell: fix background to respect theme selection and hover

Make only the normal background of the shell items transparent with a
CSS provider and leave the hovered and selected items to the theme, so
they are highlighted and follow theme changes at runtime. The overridden
normal background applied to all states and hid the highlight.

Signed-off-by: c4pp4
---
 shell/cc-shell-item-view.c | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/shell/cc-shell-item-view.c b/shell/cc-shell-item-view.c
index dc132fc..e5798fb 100644
--- a/shell/cc-shell-item-view.c
+++ b/shell/cc-shell-item-view.c
@@ -211,8 +211,19 @@ cc_shell_item_view_init (CcShellItemView *self)
   g_signal_connect (self, "button-release-event",
                     G_CALLBACK (iconview_button_release_event_cb), self);
 
-  GdkRGBA rgba = {0, 0, 0, 0};
-  gtk_widget_override_background_color (self, GTK_STATE_FLAG_NORMAL, &rgba);
+  /* Make the background transparent, but leave the hovered and
+   * selected items (.cell) to the theme */
+  GtkCssProvider *provider = gtk_css_provider_new ();
+  gtk_css_provider_load_from_data (provider,
+                                   "iconview:not(.cell),\n"
+                                   "iconview.cell:not(:selected):not(:hover) {\n"
+                                   "  background-color: transparent;\n"
+                                   "  background-image: none;\n"
+                                   "}", -1, NULL);
+  gtk_style_context_add_provider (gtk_widget_get_style_context (GTK_WIDGET (self)),
+                                  GTK_STYLE_PROVIDER (provider),
+                                  GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
+  g_object_unref (provider);
 }
 
 GtkWidget *
-- 
2.52.0

